### PR TITLE
[feature](remote)Add cache files cleaner for remote olap files

### LINKED ISSUE #11958: [Feature] Add cache files cleaner for remote olap files
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Add cache files cleaner for remote files

### Use case

Add cache files cleaner for remote files

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by pengxiangyu**
# Proposed changes

Issue Number: close #11958

## Problem summary

Add cache files cleaner for remote olap files

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/io/cache/file_cache_manager.cpp [hunk 1]
```diff
@@ -29,18 +32,110 @@ void FileCacheManager::add_file_cache(const Path& cache_path, FileCachePtr file_
...
 void FileCacheManager::remove_file_cache(const Path& cache_path) {
-    std::lock_guard<std::shared_mutex> wrlock(_cache_map_lock);
-    _file_cache_map.erase(cache_path.native());
+    bool cache_path_exist = false;
+    {
+        std::shared_lock<std::shared_mutex> rdlock(_cache_map_lock);
+        if (_file_cache_map.find(cache_path) == _file_cache_map.end()) {
+            bool cache_dir_exist = false;
+            if (global_local_filesystem()->exists(cache_path, &cache_dir_exist).ok()) {
+                if (cache_dir_exist) {
```
Comment (yangzhg) [inline be/src/io/cache/file_cache_manager.cpp:41]: redundant if stmt and var `cache_dir_exist`

File: be/src/io/cache/file_cache_manager.cpp [hunk 2]
```diff
@@ -29,18 +32,110 @@ void FileCacheManager::add_file_cache(const Path& cache_path, FileCachePtr file_
...
          iter != _file_cache_map.cend(); ++iter) {
+        if (iter->second == nullptr) {
+            continue;
+        }
         iter->second->clean_timeout_cache();
     }
 }
 
+void FileCacheManager::clean_timeout_file_not_in_mem(const Path& cache_path) {
+    std::shared_lock<std::shared_mutex> rdlock(_cache_map_lock);
```
Comment (yangzhg) [inline be/src/io/cache/file_cache_manager.cpp:71]: using `std::string_view` and `emplace_back` maybe better

File: be/src/io/cache/file_cache_manager.cpp [hunk 3]
```diff
@@ -29,18 +32,110 @@ void FileCacheManager::add_file_cache(const Path& cache_path, FileCachePtr file_
...
+    // Deal with caches not in _file_cache_map
+    if (_file_cache_map.find(cache_path.native()) == _file_cache_map.end()) {
+        std::vector<Path> cache_file_names;
+        if (io::global_local_filesystem()->list(cache_path, &cache_file_names).ok()) {
+            std::map<std::string, bool> cache_names;
+            std::list<std::string> done_names;
+            for (Path cache_file_name : cache_file_names) {
+                std::string filename = cache_file_name.native();
+                if (filename.find("_DONE") == std::string::npos) {
```
Comment (yangzhg) [inline be/src/io/cache/file_cache_manager.cpp:80]: how about a file named `/path_DONE/path/to/a/file` ?

File: be/src/io/cache/file_cache_manager.cpp [hunk 4]
```diff
@@ -29,18 +32,110 @@ void FileCacheManager::add_file_cache(const Path& cache_path, FileCachePtr file_
...
+                    cache_names[filename] = true;
+                    continue;
+                }
+                done_names.push_back(filename);
+                std::string done_file_path = (cache_path / filename).native();
+                time_t m_time;
+                if (!FileUtils::mtime(done_file_path, &m_time).ok()) {
+                    continue;
+                }
+                if (time(nullptr) - m_time < config::file_cache_alive_time_sec) {
```
Comment (yangzhg) [inline be/src/io/cache/file_cache_manager.cpp:90]: mv time(nullptr) out of for block

File: be/src/util/file_utils.cpp [hunk 5]
```diff
@@ -211,6 +211,22 @@ Status FileUtils::md5sum(const std::string& file, std::string* md5sum) {
     return Status::OK();
 }
 
+Status FileUtils::mtime(const std::string& file, time_t* m_time) {
+    int fd = open(file.c_str(), O_RDONLY);
```
Comment (yangzhg) [inline be/src/util/file_utils.cpp:215]: use a defer to protect fd maybe mor simple

File: docs/en/docs/admin-manual/config/be-config.md [hunk 6]
```diff
@@ -1576,3 +1576,22 @@ Translated with www.DeepL.com/Translator (free version)
...
 
+### `generate_cache_cleaner_task_interval_sec`
+* Type：int64
+* Description：Cleaning interval of cache files, in seconds
+* Default：43200（12 hours）
+
+### `file_cache_type`
+* Type：string
+* Description：Type of cache file. whole_ file_ Cache: download the entire segment file, sub_ file_ Cache: the segment file is divided into multiple files by size.
+* Default：104857600（100MB）
```
Comment (zxealous) [inline docs/en/docs/admin-manual/config/be-config.md:1587]: the default value is an empty string

File: docs/zh-CN/docs/admin-manual/config/be-config.md [hunk 7]
```diff
@@ -1600,3 +1600,22 @@ webserver默认工作线程数
...
 
+### `generate_cache_cleaner_task_interval_sec`
+* 类型：int64
+* 描述：缓存文件的清理间隔，单位：秒
+* 默认值：43200（12小时）
+
+### `file_cache_type`
+* 类型：string
+* 描述：缓存文件的类型。whole_file_cache：将segment文件整个下载，sub_file_cache：将segment文件按大小切分成多个文件。
+* 默认值：104857600（100MB）
```
Comment (zxealous) [inline docs/zh-CN/docs/admin-manual/config/be-config.md:1611]: default value

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**